src/control/SnappingGrid/SnappingGrid.js

```diff
@@ -351,6 +351,9 @@ class SnappingGrid extends Control {
         mapInteractions.removeAt(ourSnapInteractionLastIdx);
       }
       this.getMap().addInteraction(this.gridSnapInteraction);
+
+      // Add grid feature to grid snap interaction
+      this.gridSnapInteraction.addFeature(this.grid.getGridFeature());
```

Comment:
@symbioquine this seems to fix the issue of not snapping to the grid while drawing geometries. I feel like there should be a better solution though.

It's interesting that for both of these bugs the "fix" seems to be re-adding features each time the snap interaction is added to the map. It's almost like the snap interactions lose their features when added to the map?

Reply:
Skimming the [snap interaction code](https://github.com/openlayers/openlayers/blob/85219a86c52428e7736957e7255f45a34e645cef/src/ol/interaction/Snap.js#L411-L417) a bit, I'm wondering if your proposed change should have been required all along...

It looks like the snap interaction clears the features it is watching and re-adds them when the map its associated with changes. If the map is set to null and then subsequently set in two separate calls, I think that would result in the features the snap interaction has being lost. (Which would match the behavior you're seeing I think.)

I guess I'm saying if this proposed change fixes it, I'd say we should go for it.